stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+
+
+def get_gp_params(kernel_type, mean_type):
+    """
+    Generates a list of the parameters for the GP model based on the kernel and mean type.
+    To be used to set the order of the parameters for `get_prior` and `get_likelihood` functions.
+
+    Parameters
+    ----------
+    kernel_type: string
+        The type of kernel to be used for the Gaussian Process model
+
+    mean_type: string
+        The type of mean to be used for the Gaussian Process model
+
+    Returns
+    -------
+        A list of the parameters for the GP model
+
+    Examples
+    --------
+    get_gp_params("QPO_plus_RN", "gaussian")
+    ['log_arn', 'log_crn', 'log_aqpo', 'log_cqpo', 'log_freq', 'log_A', 't0', 'log_sig']
+    """
+    kernel_params = _get_kernel_params(kernel_type)
+    mean_params = _get_mean_params(mean_type)
+    kernel_params.extend(mean_params)
+    return kernel_params
+
+
+def get_prior(params_list, prior_dict):
+    """
+    A prior generator function based on given values.
+    Makes a jaxns specific prior function based on the given prior dictionary.
+    Jaxns requires the parameters of the prior function and log_likelihood function to
+    be in the same order. This order is made according to the params_list.
+
+    Parameters
+    ----------
+    params_list:
+        A list in order of the parameters to be used.
+
+    prior_dict:
+        A dictionary of the priors of parameters to be used.
+        These parameters should be from tensorflow_probability distributions / Priors from jaxns
+        or special priors from jaxns.
+        **Note**: If jaxns priors are used, then the name given to them should be the same as
+        the corresponding name in the params_list.
+        Also, if a parameter is to be used in the log scale, it should have a prefix of "log_"
+
+    Returns
+    -------
+    The Prior generator function.
+    The arguments of the prior function are in the order of
+        Kernel arguments (RN arguments, QPO arguments),
+        Mean arguments
+        Miscellaneous arguments
+
+    Examples
+    --------
+    A prior function for a Red Noise kernel and a Gaussian mean function
+    Obain the parameters list
+    if not can_sample:
+        pytest.skip("Jaxns not installed. Cannot make jaxns specific prior.")
+    if not tfp_available:
+        pytest.skip("Tensorflow probability required to make priors.")
+
+    params_list = get_gp_params("RN", "gaussian")
+
+    Make a prior dictionary using tensorflow_probability distributions
+    prior_dict = {
+       "log_A": tfpd.Uniform(low = jnp.log(1e-1), high = jnp.log(2e+2)),
+       "t0": tfpd.Uniform(low = 0.0 - 0.1, high = 1 + 0.1),
+       "log_sig": tfpd.Uniform(low = jnp.log(0.5 * 1 / 20), high = jnp.log(2) ),
+       "log_arn": tfpd.Uniform(low = jnp.log(0.1) , high = jnp.log(2) ),
+       "log_crn": tfpd.Uniform(low = jnp.log(1 /5), high = jnp.log(20)),
+    }
+
+    prior_model = get_prior(params_list, prior_dict)
+
```

Comment:
Could you also add an example here that uses the JaxNS priors, since you talk about them above?

Reply:
I think JaxNS prior would be rarely used and putting an example here would make things cluttered. I have added that case in the Demo notebooks.